### DISCUSSION TITLE
Page aliases for i18n. /en/about-us => /de/ueber-uns

### DISCUSSION THREAD

**Discussion by ottovw** [vercel/next.js#21181] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

The current i18n implementation works great with prefixing. However it's a problem if you want to translate the page-name itself. Which has effects on UX and SEO.
E.g.
`/en/about-us`
should be in german
`/de/ueber-uns`

## Describe the solution you'd like

To solve this, I'd like to define page-name aliases per locale.

## Describe alternatives you've considered

It can be solved with custom routing, but this breaks the easy usage of i18n.

**Comment by miguderp** [comment]
I had a similar case where I handled it through `rewrites`. It's a bit cumbersome in my opinion tho and would prefer to handle it on the page level instead of `next.config.js`

```
async rewrites() {
	return [
		{
			locale: false, // ? so we can specify /de/ underneath
			source: '/de/ueber-uns', // ? URL requested by the user
			destination: '/about-us', // ? path to your agnostic page; through the router you should use the locale for your data fetching
		},
	]
}
```

[Documentation here](https://nextjs.org/docs/api-reference/next.config.js/rewrites#rewrites-with-i18n-support)

**Reply by rnugraha** [reply]
We are using this approach as well but then wondering how do you deal with the SEO optimisation since the same content can be accessed by more than one URLs?

**Comment by mimoid-prog** [comment]
I saw this topic so many times and it's kinda sad that there is still no answer from the Next.js maintainers if they will do something about it

**Comment by jasdeepsingh** [comment]
We need capability as well. Would love to see this added/handled by default.

**Comment by crdev** [comment]
Any update now that we have NextJS 14 and it's the year 2024?
